Models/data_cleansing.py

```diff
@@ -87,14 +87,22 @@ def _create_dummies_for_categorical_features(design_matrix):
     return design_matrix
 
 
-def clean_data(design_matrix):
+def clean_data(design_matrix, remove_outliers=False):
     """Cleaning raw data before model processing."""
     design_matrix = _impute_missing_values(design_matrix)
     # design_matrix = _normalize_continuous_features(design_matrix)
     design_matrix = _create_dummies_for_categorical_features(design_matrix)
+    if remove_outliers:
+        design_matrix = outlier_treatment(design_matrix)
     return design_matrix
 
 
+def outlier_treatment(design_matrix, m=2.75):
```

Comment:
I think you can keep the value of m as it is. Just wanted to understand your thinking behind choosing the value. You're still covering more than 99% of the population, which is pretty good.

On a different note, I checked the outlier values when we keep m as 3 (or 2.5, 2.75). What we are doing here is keeping all house prices roughly above 400000 in the outlier bucket. I was thinking that due to this the predictions on test data will never go beyond 400000 figure. Now as we have 1460 data points in training data and around 10 data points are beyond 500000 mark. Assuming around same number of **actual house prices** above 500000 mark in test data as well (~ 10), as we have 1459 data points in test data. These 10 highly inaccurate predictions with range of error between 20-75% will inflate overall error as well (RMSE uses average of squares of residuals).

Should we start worrying about these cases? For example look at these two cases:
**CASE 1: Predictions for last 2 data points are bad resulting in spoiling numbers for RMSE**
![image](https://cloud.githubusercontent.com/assets/16475306/23339436/6f9db6be-fc48-11e6-9bfd-86a3e13c4377.png)

**CASE 2: Predictions for last 2 data points are better if we move our outlier threshold a bit in training data**
![image](https://cloud.githubusercontent.com/assets/16475306/23339439/7b2fb8ce-fc48-11e6-8c28-d5a968adccbf.png)

So going by this logic, we may think of deciding such a threshold of outliers which would hurt us less. Would like to hear your views as well.